### DISCUSSION TITLE
Is the a style category for horizontal dividers?

### DISCUSSION THREAD

**Discussion by jodros** [Omikhleia/resilient.sile#115] [Q&A] [answered]
The goal is to change the before/after skips.

**Comment by Omikhleia** [comment, marked as answer]
There's already so many ways it can be achieved with custom styles, that I am unsure we need a predefined style here.

Say, add this in your style file:

~~~
MyDivider:
  style:
    paragraph:
      after:
        skip: "1cm"
      before:
        skip: "1cm"
      align: "center"
~~~

Then:

~~~
## Normal divider

Lorem...

---

Lorem...

## Custom-styled div with inside divider

Lorem...

{custom-style="MyDivider"}
:::
---
:::

Lorem...

## Custom-styled divider via symbol shortcut

[^:---divider---:]: {custom-style="MyDivider"}
  :::
  ---
  :::

Lorem...

:---divider---:

Ipsum...

## Custom-styled divider via symbol registration

```=sile-lua
-- Of course one would put that in a class or package ideally
local self = SILE.documentState.documentClass
self.packages["markdown.commands"]:registerSymbol("---other---", true, function (opts)
    return {
      SU.ast.createCommand("style:apply:paragraph", { name = "MyDivider" }, {
        "• • • • • •"
      })
    }
end)
```

Lorem...

:---other---:

Lorem...
~~~

![image](https://github.com/user-attachments/assets/632b3e9d-ff09-40a7-bbd1-55063978d44a)

Each technique has pros and cons
 - (2) is quite direct for one casual use, but a bit cumbersome if repeated a lot.
 - (3) is the same shortcut idea as in https://github.com/Omikhleia/markdown.sile/discussions/136#discussioncomment-11808756 but it's only valid in a given document, so  you'd have to repeat it in other documents.
 - (4) shows programmatic symbol registration, and once registered it's available in all documents. It requires some Lua code somewhere, though ;)

BTW, there's a small bug somewhere, the "div" wrapping in (2) and (3) should not have been necessary i.e. the following should have worked:

~~~
{custom-style="MyDivider"}
---
~~~

Feel free to open a bug on markdown.sile, and perhaps analyse what I missed there and submit a PR for it ;-)

**Reply by Omikhleia** [reply]
@jodros Any feedback on the proposed answer just above?

**Reply by Omikhleia** [reply]
ping @jodros

**Reply by jodros** [reply]
Yes, this is what I was looking for.
Sorry for the delay.
